yt/yt/tests/integration/controller/test_map_operation.py

```diff
@@ -819,6 +819,31 @@ def test_row_sampling(self):
         variation = sampling_rate * (1 - sampling_rate)
         assert sampling_rate - variation <= actual_rate <= sampling_rate + variation
 
+    @authors("achulkov2")
+    def test_truncate_composite_partitioning_sample(self):
+        max_sample_size = 64 * 1024 # 64KiB
+        big_size = 15 * 1024 * 1024 # 15MiB, a bit smaller than the default row limit.
+
+        schema = [{"name": "data", "type_v3": tuple_type(["string", optional_type("int64"), list_type("string")])}]
+
+        create("table", "//tmp/t_in", attributes={"schema": schema})
+        create("table", "//tmp/t_out", attributes={"schema": schema})
+
+        # The last two strings are pretty big and definitely shouldn't be in the sample.
+        data = [{"data": ["is", 42, ["the", "answer", "to", "life" * (max_sample_size // 8), "the" * (max_sample_size // 9), "universe" * (max_sample_size // 32), "and" , "everything" * (big_size // 10)]]}]
+
+        write_table("//tmp/t_in", data, verbose=False)
+        map(in_="//tmp/t_in", out="//tmp/t_out", command="cat", spec={"max_failed_job_count": 1})
+
+        assert read_table("//tmp/t_out") == data
+
+        chunk_ids = get("//tmp/t_out/@chunk_ids")
+        assert len(chunk_ids) == 1
+
+        # Previously the whole string would end up in the sample, which would make a 10+ megabyte-sized meta.
```

Comment:
I doubt if this comment is useful outside of the review process

Reply:
Removed it.